src/api/spec/controllers/webui/cloud/azure/configurations_controller_spec.rb

```diff
@@ -0,0 +1,92 @@
+require 'rails_helper'
+
+# WARNING: Some tests require real backend answers, so make sure you uncomment
+# this line and start a test backend.
+# CONFIG['global_write_through'] = true
+
+RSpec.describe Webui::Cloud::Azure::ConfigurationsController, type: :controller, vcr: true do
+  let!(:user) { create(:confirmed_user, login: 'tom') }
+
+  before do
+    login(user)
+  end
+
+  describe 'GET #show' do
+    context 'without Azure configuration' do
+      before do
+        Feature.run_with_activated(:cloud_upload, :cloud_upload_azure) do
+          get :show
+        end
+      end
+
+      it 'creates an Azure configuration' do
+        expect(assigns(:azure_configuration)).not_to be_nil
+      end
+    end
+
+    context 'with Azure configuration' do
+      let!(:azure_configuration) { create(:azure_configuration, user: user) }
+
+      before do
+        Feature.run_with_activated(:cloud_upload, :cloud_upload_azure) do
+          get :show
+        end
+      end
+
+      after do
+        azure_configuration.destroy!
+      end
+
+      it { expect(assigns(:azure_configuration)).to eq(azure_configuration) }
```

Comment:
You factory is producing the same configuration as your callback in the controller. How do you know which is which?